src/main/java/mate/controller/DisplayAllCarsController.java

```diff
@@ -0,0 +1,37 @@
+package mate.controller;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.dao.CarDao;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Driver;
+import mate.model.Manufacturer;
+import mate.service.CarService;
+
+public class DisplayAllCarsController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final CarDao dao = (CarDao) injector.getInstance(CarDao.class);
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Car> cars = carService.getAll();
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        List<Driver> drivers = null;
+        for (Car car : cars) {
+            manufacturers.add(car.getManufacturer());
+            drivers = car.getDrivers();
+        }
+        req.setAttribute("drivers", drivers);
+        req.setAttribute("manufacturers", manufacturers);
+        req.setAttribute("cars", cars);
```

Comment:
Sorry, but I really don't understand what's going on here.
We have full cars data obtained from carService.getAll() method but instead of passing it as is to jsp page to be properly displayed we starting to extract included properties to additional list, same time drivers are overridden in cycle instead of being added...
I offer to rework jsp page to work with just list of cars and get all required info from there